head.go

```diff
@@ -402,9 +402,11 @@ func (a *headAppender) AddFast(ref uint64, t int64, v float64) error {
 	if s == nil {
 		return errors.Wrap(ErrNotFound, "unknown series")
 	}
+	s.Lock()
 	if err := s.appendable(t, v); err != nil {
 		return err
```

Comment:
Appears you're not releasing the mutex in the event of an error here.

Reply:
Thanks!